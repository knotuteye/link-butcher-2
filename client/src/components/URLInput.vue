<template>
  <form class="url-input" :action="`javascript:${updateTuple(url)}`">
    <input
      type="url"
      v-model="url"
      name="url"
      placeholder="Paste a link to shorten it"
    />
    <input type="submit" value="Shrink" />+
  </form>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator'
import { getNewLink } from '../api/calls'

@Component
export default class URLInput extends Vue {
  public url = ''
  async updateTuple(url: string) {
    const tuple = await getNewLink(`${window.location.href}slugs/create`, url)
    this.$store.commit('updateNewLink', tuple)
    console.log(tuple)
  }
}
</script>

<style scoped>
form {
  display: flex;
}
input {
  padding: 12px;
  font-size: 1.2em;
  font-family: 'Roboto', sans-serif;
  border: 0;
}

input:-webkit-autofill,
input:-webkit-autofill:hover,
input:-webkit-autofill:focus {
  background: transparent;
}

input[type='url'] {
  color: #ffffff;
  background-color: transparent;
  border: 2px solid #ffffff67;
  border-top-left-radius: 10px;
  border-bottom-left-radius: 10px;
  flex: 1;
}
input[type='url']::placeholder {
  color: #ffffffcb;
}

input[type='submit'] {
  background-color: #ffffff69;
  color: #ffffff;
  padding: 14px 2rem;
  border-top-right-radius: 10px;
  border-bottom-right-radius: 10px;
  cursor: pointer;
}

input[type='submit']:hover {
  background-color: #ffffffcb;
  color: #6f16ff;
}

input[type='submit']:focus {
  background-color: #28a018cb;
  color: #ffffff;
}

@media only screen and (max-width: 35rem) {
  form {
    display: grid;
    row-gap: 1rem;
  }
  input {
    width: 90%;
    border-radius: 10px;
  }
  input[type='submit'] {
    width: 100%;
  }
}
</style>
