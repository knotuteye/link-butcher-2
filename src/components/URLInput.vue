<template>
  <form class="url-input">
    <input type="url" name="url" placeholder="Paste a link to shorten it" />
    <input type="submit" value="Shorten" />
  </form>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator'

@Component
export default class URLInput extends Vue {}
</script>

<style scoped>
form {
  width: 75%;
  display: flex;
  max-width: 35rem;
}
input {
  padding: 12px;
  border: 0;
}
input[type='url'] {
  background-color: transparent;
  border-bottom: 2px solid #ffffff67;
  flex: 1;
}
input[type='url']::placeholder {
  color: #ffffff;
}

input[type='submit'] {
  background-color: #ffffff67;
  color: #ffffff;
  padding: 14px;
}
</style>
