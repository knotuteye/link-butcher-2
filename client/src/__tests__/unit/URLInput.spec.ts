import { mount } from '@vue/test-utils'
import URLInput from '../../components/URLInput.vue'

describe('RecentLink.vue', () => {
  test('renders as designed', () => {
    const wrapper = mount(URLInput)
    expect(wrapper.element).toMatchSnapshot()
  })
})
