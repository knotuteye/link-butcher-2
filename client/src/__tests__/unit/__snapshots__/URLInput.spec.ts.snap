// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecentLink.vue renders as designed 1`] = `
<form
  class="url-input"
  novalidate="novalidate"
>
  <input
    aria-label="Shrink the pasted link"
    type="submit"
    value="Shrink"
  />
   
  <input
    aria-label="Paste a link here"
    class=""
    name="url"
    placeholder="Paste a link to shorten it"
    required="required"
    spellcheck="false"
    type="url"
  />
</form>
`;
